setuptools/command/easy_install.py

```diff
@@ -711,9 +711,9 @@ def easy_install(self, spec, deps=False):
             else:
                 return self.install_item(spec, dist.location, tmpdir, deps)
 
-    def install_item(self, spec, download, tmpdir, deps, install_needed=False):
+    def install_item(self, spec, download, tmpdir, deps, install_needed: bool = False):
         # Installation is also needed if file in tmpdir or is not an egg
-        install_needed = install_needed or self.always_copy
+        install_needed = install_needed or bool(self.always_copy)
```

Comment:
Either this, or:
- default `always_copy` to `False` (I wasn't sure if `None` was used as a special meaning "not set by user");
- allow `install_needed` param to be `None`;
- use a differently named variable in `install_item` (like `_install_needed`);